### ISSUE TITLE
filter function often does not work

### ISSUE THREAD

**Issue by qqdaiyu55** [webtorrent/bittorrent-tracker#270]
```
filter: function (infoHash, params, cb) {
    console.log(infoHash, params.event, params.peer_id, params.downloaded, params.uploaded)

    // filter users
    // Here:   var User = mongoose.model('User', new mongoose.Schema(...))
    const userId = params.peer_id.substring(16, 40)
    User.findById(userId, function(err, user) {
      if (err) throw err
      if (user == null) cb(new Error('disallowed user ' + userId))
      else cb(null)
    })
  }
```
Sometimes it works, but often it does not.

**Comment by feross** [comment]
You might want to remove that `throw` since then you're not calling the callback.

It's not possible to debug this issue without additional information. If this is still an issue, please leave a comment or open a new issue.

**Comment by qqdaiyu55** [comment]
```
const Server = require('bittorrent-tracker').Server
const mongoose = require('mongoose')

// connect to MongoDB
mongoose.connect('mongodb://localhost/xxx')
var User = mongoose.model('User', new mongoose.Schema({
  username: {
    type: String,
    index: { unique: true }
  },
  password: String,
  avatar_url: String,
  tags: Array,
  upload: Number,
  download: Number,
  video_list: [mongoose.Schema.Types.ObjectId]
}))

var server = new Server({
  http: false, // enable http server? [default=true]
  udp: true, // enable udp server? [default=true]
  ws: true, // enable websocket server? [default=true]
  stats: true, // enable web-based statistics? [default=true]
  filter: function (infoHash, params, cb) {
    console.log(infoHash, params.event, params.peer_id, params.downloaded, params.uploaded)

    // filter (registered) users
    const userId = params.peer_id.substring(16, 40)
    User.findById(userId, function(err, user) {
      if (err || user == null) cb(new Error('disallowed user ' + userId))
      else cb(null)
    })
  }
})

server.on('error', function (err) {
  // fatal server error!
  console.log(err.message)
})

server.on('warning', function (err) {
  // client sent bad data. probably not a problem, just a buggy client.
  console.log(err.message)
})

server.on('listening', function () {
  // fired when all requested servers are listening
  console.log('listening on port:' + server.http.address().port)
})

// listen for individual tracker messages from peers:

server.on('start', function (addr) {
  console.log('got start message from ' + addr)
})

server.on('complete', function (addr) {
  console.log('complete', addr)
})
server.on('update', function (addr) {
  console.log('update', addr)
})
server.on('stop', function (addr) {
  console.log('stop', addr)
})

// start tracker server listening! Use 0 to listen on a random free port.
server.listen(6969)
```
@feross 
bittorrent-tracker: 9.7.0

I have tested for 2 days, the filter function often does not work. I mean, the filter function is not called every time a new client connects to tracker.

More details:
Sometimes a new client (transmission, utorrent, wd etc.) connects to tracker, it only prints 
```
got start message from [ip]:[port]
```
and no `console.log` information in filter function
```
console.log(infoHash, params.event, params.peer_id, params.downloaded, params.uploaded)
```

**Comment by DiegoRBaquero** [comment]
The issue relies on the fact that the filter function is only called then the swarm is created, not on every peer connection. See https://github.com/webtorrent/bittorrent-tracker/blob/master/server.js#L671

**Comment by feross** [comment]
Fixed by https://github.com/webtorrent/bittorrent-tracker/pull/272https://github.com/webtorrent/bittorrent-tracker/pull/272

**Comment by feross** [comment]
Released as 9.9.0.
